packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1678,6 +1697,24 @@ class ThemeData with Diagnosticable {
   )
   final AndroidOverscrollIndicator? androidOverscrollIndicator;
 
+  /// Obsolete property that was used for input validation errors, e.g. in
+  /// [TextField] fields. Use [ColorScheme.error] instead.
+  @Deprecated(
+    'Use colorScheme.error instead. '
+    'This feature was deprecated after v3.3.0-0.5.pre.',
+  )
+  Color get errorColor => _errorColor!;
+  final Color? _errorColor;
```

Comment:
I may have missed something here, but why do we need the private _errorColor here?

Reply:
This is because these were previously `required`, so following the pattern from https://github.com/flutter/flutter/pull/102107 for deprecated API